app/code/Magento/Customer/etc/frontend/events.xml

```diff
@@ -29,4 +29,7 @@
         <observer name="customer_password" instance="Magento\Customer\Observer\UpgradeCustomerPasswordObserver" />
         <observer name="customer_unlock" instance="Magento\Customer\Observer\CustomerLoginSuccessObserver" />
     </event>
+    <event name="customer_save_after_data_object">
```

Comment:
Please check if we need to update email if it was changed from the backend. Right now you have this observer at the frontend area.

Reply:
Check and fix for backend too